sdk/eventhub/azure-eventhub/azure/eventhub/amqp/_amqp_message.py

```diff
@@ -237,54 +235,16 @@ def _from_amqp_message(self, message):
         self._annotations = message.message_annotations if message.message_annotations else {}
         self._delivery_annotations = message.delivery_annotations if message.delivery_annotations else {}
         self._application_properties = message.application_properties if message.application_properties else {}
-
-    def _to_outgoing_amqp_message(self):
-        message_header = None
-        if self.header and any(self.header.values()):
-            message_header = Header(
-                delivery_count=self.header.delivery_count,
-                ttl=self.header.time_to_live,
-                first_acquirer=self.header.first_acquirer,
-                durable=self.header.durable,
-                priority=self.header.priority
-            )
-
-        message_properties = None
-        if self.properties and any(self.properties.values()):
-            message_properties = Properties(
-                message_id=self.properties.message_id,
-                user_id=self.properties.user_id,
-                to=self.properties.to,
-                subject=self.properties.subject,
-                reply_to=self.properties.reply_to,
-                correlation_id=self.properties.correlation_id,
-                content_type=self.properties.content_type,
-                content_encoding=self.properties.content_encoding,
-                creation_time=int(self.properties.creation_time) if self.properties.creation_time else None,
-                absolute_expiry_time=int(self.properties.absolute_expiry_time)
-                if self.properties.absolute_expiry_time else None,
-                group_id=self.properties.group_id,
-                group_sequence=self.properties.group_sequence,
-                reply_to_group_id=self.properties.reply_to_group_id
-            )
-
-        dict = {
-            "header": message_header,
-            "properties":  message_properties,
-            "application_properties": self.application_properties,
-            "message_annotations": self.annotations,
-            "delivery_annotations": self.delivery_annotations,
-            "footer": self.footer
-        }
-
-        if self.body_type == AmqpMessageBodyType.DATA:
-            dict["data"] = self._body
-        elif self.body_type == AmqpMessageBodyType.SEQUENCE:
-            dict["sequence"] = self._body
+        if message.data:
+            self._body = message.data
```

Comment:
when message.data is set to self._body, self._body only returns the data when called the first time. Not sure why, b/c calling message.data returns data every time.

Reply:
solution: reading into list